ros2multicast/test/test_api.py

```diff
@@ -12,24 +12,99 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import random
+import sys
 import threading
 import time
 
+import pytest
+
 from ros2multicast.api import receive
 from ros2multicast.api import send
 
 
-def test_api():
-    sent_data = b'test_api'
+def _send_receive(sent_data, rx_kwargs, tx_kwargs):
     received_data = None
 
     def target():
         nonlocal received_data
-        received_data, _ = receive(timeout=1.0)
+        try:
+            received_data, _ = receive(**rx_kwargs)
+        except TimeoutError:
+            pass
 
     t = threading.Thread(target=target)
     t.start()
     time.sleep(0.1)
-    send(sent_data)
+    send(sent_data, **tx_kwargs)
     t.join()
-    assert sent_data == received_data
+    return received_data
+
+
+def test_api():
+    sent_data = b'test_api'
+
+    rx_kwargs = {'timeout': 1.0}
+    tx_kwargs = {}
+
+    assert sent_data == _send_receive(sent_data, rx_kwargs, tx_kwargs)
+
+
+def test_group_and_port():
+    sent_data = b'test_api'
+
+    r1 = random.randrange(0, 255)
+    r2 = random.randrange(0, 255)
+    r3 = random.randrange(1, 255)
+    port = random.randrange(16000, 18000)
+
+    group = f'234.{r1}.{r2}.{r3}'
+
+    tx_kwargs = {'group': group, 'port': port}
+    rx_kwargs = {'timeout': 1.0}
+    rx_kwargs.update(tx_kwargs)
+
+    assert sent_data == _send_receive(sent_data, rx_kwargs, tx_kwargs)
+
+
+def test_group_mismatch():
+    sent_data = b'test_api'
+
+    r1 = random.randrange(0, 255)
+    r2 = random.randrange(0, 255)
+    r3 = random.randrange(1, 255)
+    port = random.randrange(16000, 18000)
+
+    group1 = f'234.{r1}.{r2}.{r3}'
+    group2 = f'235.{r1}.{r2}.{r3}'
+
+    tx_kwargs = {'group': group1, 'port': port}
+    rx_kwargs = {'group': group2, 'port': port, 'timeout': 1.0}
+
+    try:
+        assert _send_receive(sent_data, rx_kwargs, tx_kwargs) is None
+    except OSError as e:
+        if sys.platform.startswith('win'):
+            if 10051 == e.winerror:
+                # TODO(sloretz) understand why this test fails this way in CI
+                # "A socket operation was attempted to an unreachable network"
+                pytest.skip('Unknown why this OSError occurs on Windows')
+        else:
+            raise
```

Comment:
I'm unable to reproduce this error in my windows VM. I tried both running the test, and had a `while` loop send multicast packets to random groups and ports for a while.

I'm assuming it is specific to the network configuration of the CI machines, and skipping the test when this error occurs.